compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java

```diff
@@ -36,14 +38,18 @@
     private final BuildOptions buildOptions;
     private final ProjectEnvironment projectEnvironment;
     private final ProjectKind projectKind;
+    private final Target target;
 
     protected Project(ProjectKind projectKind,
                       Path projectPath,
-                      ProjectEnvironmentBuilder projectEnvironmentBuilder, BuildOptions buildOptions) {
+                      ProjectEnvironmentBuilder projectEnvironmentBuilder,
+                      Target target,
```

Comment:
Passing the target path via this `Target` object is not the correct way IMO. We should allow the user to configure such things via the ProjectEnvironmentBuilder.